pkg/services/ngalert/api/api_ruler.go

```diff
@@ -0,0 +1,229 @@
+	log   log.Logger
+}
+
+func (srv RulerSrv) RouteDeleteNamespaceRulesConfig(c *models.ReqContext) response.Response {
+	namespace := c.Params(":Namespace")
+	namespaceUID, err := srv.store.GetNamespaceUIDBySlug(namespace, c.SignedInUser.OrgId, c.SignedInUser)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, fmt.Sprintf("failed to get namespace: %s", namespace), err)
+	}
+	if err := srv.store.DeleteNamespaceAlertRules(&ngmodels.DeleteNamespaceAlertRulesCommand{
+		OrgID:        c.SignedInUser.OrgId,
+		NamespaceUID: namespaceUID,
+	}); err != nil {
+		return response.Error(http.StatusInternalServerError, "failed to delete namespace alert rules", err)
+	}
+	return response.JSON(http.StatusAccepted, util.DynMap{"message": "namespace rules deleted"})
+}
+
+func (srv RulerSrv) RouteDeleteRuleGroupConfig(c *models.ReqContext) response.Response {
+	namespace := c.Params(":Namespace")
+	namespaceUID, err := srv.store.GetNamespaceUIDBySlug(namespace, c.SignedInUser.OrgId, c.SignedInUser)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, fmt.Sprintf("failed to get namespace: %s", namespace), err)
+	}
+	ruleGroup := c.Params(":Groupname")
+	if err := srv.store.DeleteRuleGroupAlertRules(&ngmodels.DeleteRuleGroupAlertRulesCommand{
+		OrgID:        c.SignedInUser.OrgId,
+		NamespaceUID: namespaceUID,
+		RuleGroup:    ruleGroup,
+	}); err != nil {
+		return response.Error(http.StatusInternalServerError, "failed to delete group alert rules", err)
+	}
+	return response.JSON(http.StatusAccepted, util.DynMap{"message": "rule group deleted"})
+}
+
+func (srv RulerSrv) RouteGetNamespaceRulesConfig(c *models.ReqContext) response.Response {
+	namespace := c.Params(":Namespace")
+	namespaceUID, err := srv.store.GetNamespaceUIDBySlug(namespace, c.SignedInUser.OrgId, c.SignedInUser)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, fmt.Sprintf("failed to get namespace: %s", namespace), err)
+	}
+
+	q := ngmodels.ListNamespaceAlertRulesQuery{
+		OrgID:        c.SignedInUser.OrgId,
+		NamespaceUID: namespaceUID,
+	}
+	if err := srv.store.GetNamespaceAlertRules(&q); err != nil {
+		return response.Error(http.StatusInternalServerError, "failed to update rule group", err)
+	}
+
+	result := apimodels.NamespaceConfigResponse{}
+	ruleGroupConfigs := make(map[string]apimodels.GettableRuleGroupConfig)
+	for _, r := range q.Result {
+		ruleGroupConfig, ok := ruleGroupConfigs[r.RuleGroup]
+		switch ok {
+		case false:
+			ruleGroupInterval := model.Duration(time.Duration(r.IntervalSeconds) * time.Second)
+			ruleGroupConfigs[r.RuleGroup] = apimodels.GettableRuleGroupConfig{
+				Name:     r.RuleGroup,
+				Interval: ruleGroupInterval,
+				Rules: []apimodels.GettableExtendedRuleNode{
+					toGettableExtendedRuleNode(*r),
+				},
+			}
+		case true:
+			ruleGroupConfig.Rules = append(ruleGroupConfig.Rules, toGettableExtendedRuleNode(*r))
+			ruleGroupConfigs[r.RuleGroup] = ruleGroupConfig
+		}
```

Comment:
I think this is much better written as a normal `if` statement.
```suggestion
		if !ok {
			ruleGroupInterval := model.Duration(time.Duration(r.IntervalSeconds) * time.Second)
			ruleGroupConfigs[r.RuleGroup] = apimodels.GettableRuleGroupConfig{
				Name:     r.RuleGroup,
				Interval: ruleGroupInterval,
				Rules: []apimodels.GettableExtendedRuleNode{
					toGettableExtendedRuleNode(*r),
				},
			}
		} else {
			ruleGroupConfig.Rules = append(ruleGroupConfig.Rules, toGettableExtendedRuleNode(*r))
			ruleGroupConfigs[r.RuleGroup] = ruleGroupConfig
		}
```